hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -518,6 +518,58 @@ public static boolean validateChecksum(Properties props) {
     return Long.parseLong(props.getProperty(TABLE_CHECKSUM.key())) == generateChecksum(props);
   }
 
+  /**
+   * This function returns the partition fields joined by BaseKeyGenerator.FIELD_SEPARATOR. It will also
+   * include the key generator partition type with the field. The key generator partition type is used for
+   * Custom Key Generator.
+   */
+  public static Option<String> getPartitionFieldPropForKeyGenerator(HoodieConfig config) {
```

Comment:
For all these static methods, do we have unit test covering different combinations of `PARTITION_FIELDS` (including complex and simple)?

Reply:
I have added the tests in the PR handling partition schema

Reply:
Moved the test to this PR